src/mono/mono/mini/method-to-ir.c

```diff
@@ -5756,8 +5756,10 @@ check_get_virtual_method_assumptions (MonoClass* klass, MonoMethod* method)
  * Returns null, if the optimization cannot be performed.
  */
 static MonoMethod*
-try_prepare_objaddr_callvirt_optimization (MonoCompile *cfg, guchar *next_ip, guchar* end, MonoMethod *method, MonoGenericContext* generic_context, MonoClass *klass)
+try_prepare_objaddr_callvirt_optimization (MonoCompile *cfg, guchar *next_ip, guchar* end, MonoMethod *method, MonoGenericContext* generic_context, MonoType *param_type)
 {
+	MonoClass *klass = mono_class_from_mono_type_internal (param_type);
```

Comment:
I'm not sure why the old code checked for `!klass` but we should not pass a null param_type to "class from mono type" so maybe bail if param_type is null just in case?